pkg/resources/statefulsets/minio-statefulset.go

```diff
@@ -390,5 +390,20 @@ func NewForMinIO(mi *miniov1.MinIOInstance, serviceName string, hostsTemplate st
 			ss.Spec.VolumeClaimTemplates = append(ss.Spec.VolumeClaimTemplates, pvClaim)
 		}
 	}
+
+	// configure the pod to register into disco
+	if ss.Spec.Template.ObjectMeta.Annotations == nil {
+		ss.Spec.Template.ObjectMeta.Annotations = map[string]string{}
+	}
+	ss.Spec.Template.ObjectMeta.Annotations["io.min.disco"] = fmt.Sprintf("{.metadata.name}.%s", mi.HostPostfix())
+	// if a Disco Service IP was provided, global service discovery is not enabled, so we are going to configure the
```

Comment:
This is to speed up Pod discovery, based on recent testing by @harshavardhana this is not needed. Also, that should be a separate PR.

Reply:
what do you mean it's not needed? the fallback is needed to guarantee the service discovery always works for MinIO even if it has not been configured globally

Reply:
As far as I know, nodes currently talk to each other using node names which is resolved using standard dns configuration for stateful set node names. Minio is not aware of how it is presented as a single service.

Reply:
This will allow minio pods to resolve the Bucket DNS names? Is there another purpose for it?